consensus/src/detail/mod.rs

```diff
@@ -142,30 +143,53 @@ impl Consensus {
         }
     }
 
+    /// returns the new block index, which is the new tip, if any
+    fn process_orphans(&mut self, last_processed_block: &Id<Block>) -> Option<BlockIndex> {
+        let orphans = self.orphan_blocks.take_all_children_of(last_processed_block);
+        let orphan_processing_result = orphans
+            .into_iter()
+            .map(|blk| self.process_block(blk, BlockSource::Local))
+            .collect::<Vec<_>>();
+
+        let (block_indexes, block_errors): (Vec<Option<BlockIndex>>, Vec<BlockError>) =
+            orphan_processing_result.into_iter().partition_result();
```

Comment:
I guess you no longer need to create the intermediate `Vec`

Reply:
Unless you mean the previous vector, in which case, clippy is already yelling at me 😄